src/instrumentation/do-storage.ts

```diff
@@ -41,25 +63,116 @@ const StorageAttributes: Record<string | symbol, ExtraAttributeFn> = {
 			}
 		}
 		if (argArray.length > 1) {
-			Object.assign(attrs, argArray[1])
+			const options = argArray[1] as DurableObjectPutOptions
+			if ('allowConcurrency' in options) {
+				attrs['db.cf.do.allow_concurrency'] = options.allowConcurrency
+			}
+			if ('noCache' in options) {
+				attrs['db.cf.do.no_cache'] = options.noCache
+			}
 		}
 		return attrs
 	},
 	list(argArray, result: Map<string, unknown>) {
```

Comment:
This is out of scope of this PR, but I wonder if we should try to add types to `argArray` in our proxies?